src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -58,6 +58,29 @@ public void SetEncoding(bool value)
         int IVsEditorFactory.Close()
             => VSConstants.S_OK;
 
+        public int ChooseEditorFactory(
+            string pszMkDocument,
+            IVsHierarchy pHier,
+            uint itemid,
+            IntPtr punkDocDataExisting,
+            ref Guid rguidLogicalView,
+            out Guid pguidEditorTypeActual,
+            out Guid pguidLogicalViewActual)
+        {
+            if (rguidLogicalView == VSConstants.LOGVIEWID.Designer_guid)
+            {
+                pguidEditorTypeActual = typeof(IWinFormsEditorFactory).GUID;
+                pguidLogicalViewActual = rguidLogicalView;
+                return VSConstants.S_OK;
+            }
```

Comment:
can we doc what's going on here.  i don't have any good sense for why roslyn is responding to a request to make the editor factory for , and why we're then saying it's for winforms to handle :)

Reply:
Since a WinForms' file (Form.cs|.vb) has same file extension as any C# or VB file, Roslyn's `AbstractEditorFactory` gets called for it as well.
If `PhysicalView` is of type `Form`, `AbstractEditorFactory` makes a call to  `IWinFormsEditorFactory.CreateEditorInstance` to load the designer.